Automation for LabKey R&D UX - Field Editor Updates 2020, Summary View

labkey-sweta opened the pull request:
#### Rationale
1. Adding test coverage for toggle between Summary and detail mode
2. Links in the detail mode

#### Related Pull Requests
* https://github.com/LabKey/sampleManagement/pull/503
* https://github.com/LabKey/labkey-ui-components/pull/464
* https://github.com/LabKey/platform/pull/2066

#### Changes

1. src/org/labkey/test/components/domain/DomainFormPanel.java
2. src/org/labkey/test/tests/FieldEditorRowSelectionActionTest.java

labkey-sweta committed:
Automation for LabKey R&D UX - Field Editor Updates 2020, Summary View

RosalineP commented on src/org/labkey/test/tests/FieldEditorRowSelectionActionTest.java:
```diff
@@ -27,10 +26,14 @@
 public class FieldEditorRowSelectionActionTest extends BaseWebDriverTest
 {
     private final static String PROJECT_NAME = "Field Editor Row Selection Action Test";
+    private final static List<String> expectedHeaders = Arrays.asList("Name", "Range URI", "Required", "Lock Type", "Lookup Container", "Lookup Schema", "Lookup Query",
```
Should we include a test in a folder in which the `Ontology` module is not enabled, in order to ensure `"Source Ontology", "Concept Import Column", "Concept Label Column", "Principal Concept Code"` columns accurately do not get rendered, or would that be an overzealous test?

labkey-sweta replied:
Done.

RosalineP commented on src/org/labkey/test/tests/FieldEditorRowSelectionActionTest.java:
```diff
@@ -83,6 +88,19 @@ public void testListActions() throws Exception
 
         checker().verifyTrue("Exported Fields are not same as UI", uiFields.equals(exportedFields));
 
+        log("Verifying the toggle between summary view and detail view list domain");
```
The section here, and the block at line 127, and at 200, have a lot of similarities and the differences they do have look like they could be adjusted via function arguments, if they were modularized into a function `testSummaryViewActions` or something.

Would it be good style to group these three chunks into one function, or in tests is it preferable to have the code be more explicit like this?

labkey-sweta replied:
I thought about it, it was mostly assert statements and not a lot of action having. Decided to keep them in the test.

Let me know if you feel strongly to move.

RosalineP reviewed (commented):
Just a couple comments! Thanks for doing the tests on this 👍

Please leave merge task to me, as I will be going through the ui-components publishing process on these branches due to Product drift necessitating a Product update between feature merge and Automation Test merge.

labkey-sweta committed:
1. Disabling Ontology for DatasetAction test and verifying the columns enabling it back for project.

RosalineP reviewed (approved):
Looks good!

labkey-sweta committed:
Updates

1. New automation test for ontology
2. Remove ontology reference from the FieldEditorRowSelectionActionTest

RosalineP committed:
Merge branch 'develop' into fb_automationSummaryView
